### PR TITLE
WIP: Windows Arm64 port of binutils

### PR DESCRIPTION
**Pull request by Blackhex**
Pre-liminary WIP PR building binutils MSYS2 package for Windows Arm64 using https://github.com/ZacWalk/binutils-woarm64.git fork.

Contributes to #3834 .

### CODE REVIEW

**Comment by Biswa96** [comment]
Packages are only built using upstream source repository. Please upstream the changes first.

**Comment by Blackhex** [comment]
Hello @Biswa96. Sure, we want to upstream the changes before this PR will be submitted for the review. But it's chicken-egg problem - we will have a hard time to upstream them unless we know they are working, e.g. they can compile bash.exe.

File: binutils/PKGBUILD [hunk 1]
```diff
@@ -1,30 +1,45 @@
...
             '4e8ac055df61b1b5d6ae29dc87e1154737c2e87c7b244b44866702cabf1a5d18'
             '2acce83fb67a32e2cd54708ca97c31c79085d0c0f5e2f42b7d103c1fba3250fc'
             '57478b9971183d430c93701b1d533e3724dab5334bbf44db924777e4a93c1063')
 validpgpkeys=('EAF1C276A747E9ED86210CBAC3126D3B4AE55E93'
               '3A24BC1E8FB409FA9F14371813FCEF89DD9E3C4F')
 
+pkgver() {
+  cd "${srcdir}/${_realname}"
+
+  printf "%s" "$(git rev-parse --short HEAD)"
```
Comment (Biswa96) [inline binutils/PKGBUILD:38]: Try to use the pkgver function from here https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-PKGBUILD-templates/PKGBUILD.CMake-git.sh

Comment (Biswa96) [reply binutils/PKGBUILD:38]: > But it's chicken-egg problem 

Just ate an egg today, didn't know it is related to binutils 😄

**Comment by lazka** [comment]
I wonder if we could make it an additional package that conflicts/provides binutils for starters.

**Comment by lazka** [comment]
> Can you please explain the reasons behind it a bit?

Sure, sorry. This is a "random" fork (we've had some contact with Zac before, so I know where this is coming from at least) based on an unstable version of binutils. We normally don't ship development binutils builds and also I don't know the state of the fork and how it might affect the stability for our current use cases (i.e. building all our packages)

We can backport those patches to the last stable release, or someone could try to convince me that this has no downsides for our day to day builds as is. Or as I suggested above we could make a separate package that users have to explicitly install, or other packages have to explicitly depend on to use the new functionality (`pacman -S binutils-woarm64` for example)

**Comment by sskras** [comment]
@Blackhex, what do you think about publishing the packages in your own package repository host on GitHub:
https://disconnected.systems/blog/archlinux-repo-in-a-git-repo/

> For aursync to work, we need to add a repository to /etc/pacman.conf
> ```
> [mdaffin]
> SigLevel = Optional TrustAll
> Server = https://raw.githubusercontent.com/mdaffin/arch-repo/master/x86_64
> ```
> 

... or maybe even using GitHub pages 
https://www.sainnhe.dev/post/create-personal-arch-linux-package-repository/

> Wait for several minutes, and your github pages will be published at `https://sainnhe.github.io/my_repo`.
> 
> So the url of package repository should be `https://sainnhe.github.io/my_repo/x86_64`.
> 
> To add this repository, add something like this at the end of `/etc/pacman.conf`:
> ```
> [sainnhe]
> Server = https://sainnhe.github.io/my_repo/x86_64
> ```
> 
> Now refresh the package databases and your package repository will be successfully added.

That would be an easy way to share you work both with testers and @msys2 crew.

**Comment by Blackhex** [comment]
> what do you think about publishing the packages in your own pacman repository hosted on GitHub
That's actually a good idea but we are very far from that at this point... We are doing just first pre-liminary experiments with the packages to identify the gaps on the codegen implementation.

**Comment by jeremyd2019** [comment]
> what do you think about publishing the packages in your own pacman repository hosted on GitHub: 

I've used a GitHub release for that, but it did expose an error in pacman once due to how they redirect to S3, but it got patched.

**Comment by Blackhex** [comment]
JFYI, I am still working on this but I've just renamed the branch as it contains not only binutils-related changes now. I'll file individual PRs as they will be ready for review.

**Pull request closed without merging**